hedera-node/hedera-networkadmin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/UnzipUtility.java

```diff
@@ -0,0 +1,137 @@
+/*
```

Comment:
Maybe worth to move this class to common area like spi because it can be used by other services.

Reply:
If it does end up being useful other places then we can move it. In the mono-service code it was only used by freeze.

Reply:
I understand but I saw a similar code that we implemented as part of the disaster recovery with Iris. most of it similar code

Reply:
Making this shared should be a separate effort done later.  Likely the functionality belongs in platform and should be very carefully rewritten to use channels and work without interfering with the MerkleDb filesystem accesses (which are extremely frequent and probably should be higher priority than unpacking this file).  Short term if there is other Services code that needs to unzip data to a file, then we should add an issue to investigate unifying the two implementations.

Reply:
ok no problem at all I just added it as general comment I didn't mentioned to do it right away.

Reply:
I will leave it here for now.

Reply:
I will leave it here for now.